java/src/main/java/com/bazaarvoice/gumshoe/Context.java

```diff
@@ -25,6 +25,7 @@
     public static enum Status {
         CREATED, STARTED, FINISHED;
     }
+    private Gumshoe gumshoe;
```

Comment:
A ```Gumshoe``` instance is how the app interacts with gumshoe.  It defines a DSL for creating/finishing/failing contexts, loading it with data, and emitting events.  It must be a thread local instance.  Example:

```java
Gumshoe gumshoe = Gumshoe.get();
gumshoe.context("my-context").put("foo", "bar").start();
gumshoe.emit("an event");
gumshoe.context().finish();
```

Contexts are created through the governing gumshoe, and it maintains the stack of contexts for the thread.  When a context finishes itself with the ```gumshoe.context().finish()``` statement above, it should pop itself off the stack of contexts in the gumshoe instance.

This PR changes things so that when a ```Gumshoe``` instance creates a ```Context```, it passes itself as a reference.  Thus when the context is eventually finished, it asks the governing ```Gumshoe``` that created it to pop its stack.